specifyweb/frontend/js_src/lib/components/Molecules/DateElement.tsx

```diff
@@ -19,14 +19,42 @@ export function DateElement({
   readonly fallback?: React.ReactNode;
   readonly flipDates?: boolean;
 }): JSX.Element {
+  const [relativeDate, setRelativeDate] = React.useState<string | undefined>(
+    date ? getRelativeDate(new Date(date)) : undefined
+  );
+
+  React.useEffect(() => {
+    let intervalValue;
+    let interval: NodeJS.Timeout | undefined;
+
+    if (date) {
```

Comment:
(optional)
can you play arround with refactoring this code to use setTimeout?
it might be simplified as a result or it might not.
but in either case it would be good exercise.
if code would be more complicated afterward, feel free to revert.

here is how that might look like:
you have a function that updates the state, and calls setTimeout with appropriate timeout value - the result of setTimeout will be stored in `interval` just like now
you call the function